### PR TITLE
concurrencpp: add 0.1.6

### PR DESCRIPTION
**Pull request by chausner**
Specify library name and version:  **concurrencpp/0.1.6**

This updates the concurrencpp port to the latest version 0.1.6.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/concurrencpp/all/conanfile.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def layout(self):
     def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             check_min_cppstd(self, self._min_cppstd)
-        if self.options.shared and is_msvc(self):
+        if Version(self.version) <= "0.1.5" and self.options.shared and is_msvc(self):
```
Comment (chausner) [inline recipes/concurrencpp/all/conanfile.py:60]: https://github.com/David-Haim/concurrencpp/issues/75 has been fixed in 0.1.6, thus shared builds on Windows are supported now.

Comment (uilianries) [reply recipes/concurrencpp/all/conanfile.py:60]: Nice, do you have any reference/issue/url indicating that fix? It would be great for traceability.

Comment (chausner) [reply recipes/concurrencpp/all/conanfile.py:60]: Yes, this is the MR where it got fixed: https://github.com/David-Haim/concurrencpp/pull/94

Comment (uilianries) [reply recipes/concurrencpp/all/conanfile.py:60]: @chausner Thank you!

**Review by uilianries — APPROVED** [review]
LGTM

**Review by jwillikers — APPROVED** [review]

**Pull request merged**